### PR TITLE
Update adaptive detail row when new row is deleted in batch edit mode has been fixed (T533190)

### PR DESCRIPTION
**Pull request by jsdmitry**

### CODE REVIEW

File: js/ui/data_grid/ui.data_grid.adaptivity.js [hunk 1]
```diff
@@ -704,8 +705,8 @@ gridCore.registerModule("adaptivity", {
                     }
                 },
 
-                _beforeUpdateItems: function(rowIndices, rowIndex) {
-                    if(this.getEditMode() !== DATAGRID_EDIT_MODE_FORM && this._adaptiveController.hasHiddenColumns()) {
+                _beforeUpdateItems: function(rowIndices, rowIndex, operation) {
```
Comment (Alyar666) [inline js/ui/data_grid/ui.data_grid.adaptivity.js]: rename _beforeUpdateItems -> _beforeEditRow

File: js/ui/data_grid/ui.data_grid.adaptivity.js [hunk 2]
```diff
@@ -723,6 +724,9 @@ gridCore.registerModule("adaptivity", {
                         rowIndices.push(rowIndex + 1);
                         this._dataController.adaptiveExpandedKey(item.key);
                     }
+                    if(operation === "delete" && this.getEditMode() === DATAGRID_EDIT_MODE_BATCH) {
```
Comment (Alyar666) [inline js/ui/data_grid/ui.data_grid.adaptivity.js]: Create a new method - _beforeDeleteRow. Write this code there

**Review by Alyar666 — DISMISSED** [review]
review

File: js/ui/data_grid/ui.data_grid.adaptivity.js [hunk 3]
```diff
@@ -788,6 +789,17 @@ gridCore.registerModule("adaptivity", {
                     this.callBase(rowIndex);
                 },
 
+                deleteRow: function(rowIndex) {
+                    var expandedKey = this._dataController._adaptiveExpandedKey,
+                        expandedRowIndex = gridCoreUtils.getIndexByKey(expandedKey, this._dataController.items());
```
Comment (vconst) [inline js/ui/data_grid/ui.data_grid.adaptivity.js]: I think that using getKeyByRowIndex and isAdaptiveDetailRowExpanded is better way then using private fields and utility method

**Review by vconst — CHANGES_REQUESTED** [review]

**Review by vconst — APPROVED** [review]

**Pull request merged**